sphinx_gallery/sorting.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+r"""
+Sorters for Sphinx-Gallery subsections
+======================================
+
+Sorting key functions for gallery subsection folders
+"""
+# Created: Sun May 21 20:38:59 2017
+# Author: Óscar Nájera
+# License: 3-clause BSD
+
+from __future__ import division, absolute_import, print_function
+import os
+import types
+
+
+class ExplicitOrderStrict(object):
+    """Sorting key for all galleries sub sections
+
+    Parameters
+    ----------
+    ordered_list : list, tuple, types.GeneratorType
+        Hold the paths of each gallery subfolder"""
+
+    def __init__(self, ordered_list):
+        if not isinstance(ordered_list, (list, tuple, types.GeneratorType)):
+            raise ValueError("ExplicitOrderStrict sorting key takes a list, "
+                             "tuple or Generator, which hold"
+                             "the paths of each gallery subfolder")
+
+        self.ordered_list = list(os.path.normpath(path)
+                                 for path in ordered_list)
+
+    def __call__(self, item):
+        if item in self.ordered_list:
+            return self.ordered_list.index(item)
+        else:
```

Comment:
and here could be an `elif self.strict is False: return np.inf` or something like that.

Though if we don't want to support that behavior it's fine with me, just don't think we need to call it `ExplicitOrderStrict` in that case since it is *always* strict